tools/OpenConsole.psm1

```diff
@@ -238,4 +238,32 @@ function Debug-OpenConsole()
     Debug-Process -Id $process.Id
 }
 
-Export-ModuleMember -Function Set-MsbuildDevEnvironment,Invoke-OpenConsoleTests,Invoke-OpenConsoleBuild,Start-OpenConsole,Debug-OpenConsole
+#.SYNOPSIS
+# runs clang-format on a file
+#
+#.PARAMETER Filename
+# The full path to the file to format
+function Invoke-ClangFormat() {
+    param (
+        [parameter(Mandatory=$true)]
+        [string]$Filename
+    )
+    & "$env:OpenconsoleRoot/dep/llvm/clang-format" -i $Filename
+}

```

Comment:
This one is a little bit more robust and supports multiple files:

```powershell
function Invoke-ClangFormat {
    [CmdletBinding()]
    Param (
        [Parameter(Mandatory=$true,ValueFromPipeline=$true)]
        [string[]]$Path
    )

    Process {
        ForEach($_ in $Path) {
            Try {
                $n = Get-Item $_ -ErrorAction Stop | Select -Expand FullName
                & clang-format -i $n
            } Catch {
                Write-Error $_
            }
        }
    }
}
```

just adjust the path to clang-format

then you can make `Invoke-CodeFormat` a lot simpler:

`gci -r "$env:OpenConsoleRoot/src" -include *.cpp,*.hpp,*.h | Where FullName -NotLike "*Generated Files*' | Invoke-ClangFormat`

Reply:
[string]$Filename
    )
    & "$env:OpenconsoleRoot/dep/llvm/clang-format" -i $Filename
}